### PR TITLE
API update - Adds SiteMonthlyMetrics viewset

### PR DESCRIPTION
**Pull request by johnbaldwin**
SiteMonthlyMetrics viewset
* Adds new endpoints to retrieve metrics currently in 'GeneralSiteMetricsViewSet'
* endpoints have underscore in url for the custom actions. Deferring
fix. Should create a bug card for this

Other fixes and updates in this commit:

* Fixed figures.metrics naive datetime objects
* Updated figures.metrics to use the User model directly for registered
users (instead of getting from SiteDailyMetrics)

### CODE REVIEW

**Comment by johnbaldwin** [comment]
Added test for MAU option in the `populate_figures_metrics` Django management command to get the 'codecov/patch' check to pass

File: figures/helpers.py [hunk 1]
```diff
@@ -110,6 +110,10 @@ def prev_day(val):
     return days_from(val, -1)
 
 
+def days_in_month(month_for):
+    return calendar.monthrange(month_for.year, month_for.month)[1]
```
Comment (OmarIthawi) [inline figures/helpers.py]: No biggie, better than having the magical constant `1`.

```suggestion
    """
    Calculate days in a certain month.
    """
    _from_day, to_day = calendar.monthrange(month_for.year, month_for.month)[1]
    return to_day
```

File: tests/test_commands.py [hunk 2]
```diff
@@ -1,15 +1,60 @@
...
+def test_mau_no_delay_with_mock(transactional_db):
+    """
+    This would be the preferred test to the above. Acccording to the documentation,
+    this should work, but the mock never gets called, so something is not right
+
+    https://docs.python.org/dev/library/unittest.mock.html#where-to-patch
+    """
+    path = 'figures.management.commands.populate_figures_metrics.populate_all_mau'
+    with mock.patch(path) as mock_populate_all_mau:
+        call_command('populate_figures_metrics', '--no-delay')
```
Comment (OmarIthawi) [inline tests/test_commands.py]: Thanks for documenting that it's not working. I tested it, and looks like the `--mau` argument is missing. This should fix it:

```suggestion
        call_command('populate_figures_metrics', '--no-delay', '--mau')
```

File: tests/test_commands.py [hunk 3]
```diff
@@ -1,15 +1,60 @@
...
     def test_command_output(self):
         out = StringIO()
         call_command('populate_figures_metrics', '--no-delay', stdout=out)
 
         self.assertEqual('', out.getvalue())
-        #self.assertIn('Expected output', out.getvalue())
\ No newline at end of file
+
+
+def test_mau_no_delay(monkeypatch):
```
Comment (OmarIthawi) [inline tests/test_commands.py]:
```suggestion
```

Thanks for documenting this, please see below. I think I've fixed the original function, so this one isn't needed.

File: tests/test_commands.py [hunk 4]
```diff
@@ -1,15 +1,60 @@
...
+        call_list.append(True)
+
+    path = 'figures.management.commands.populate_figures_metrics.populate_all_mau'
+    monkeypatch.setattr(path, mock_populate_all_mau)
+    call_command('populate_figures_metrics', '--no-delay', '--mau')
+    assert len(call_list) == 1
+
+
+@pytest.mark.skip('This test does not work')
+def test_mau_no_delay_with_mock(transactional_db):
```
Comment (OmarIthawi) [inline tests/test_commands.py]:
```suggestion
def test_mau_no_delay(transactional_db):
```

File: figures/helpers.py (hunk 1 above)
Comment (johnbaldwin) [reply figures/helpers.py]: @OmarIthawi Sure, we'll do that, but we need to drop the `[1]` from the call you suggest, and the first value is the day of the week of the first day of the month. This is the new function:

```
def days_in_month(month_for):
    _, num_days_in_month = calendar.monthrange(month_for.year, month_for.month)
    return num_days_in_month
```

File: tests/test_commands.py (hunk 2 above)
Comment (johnbaldwin) [reply tests/test_commands.py]: Thanks @OmarIthawi !

**Pull request merged**